source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -394,6 +400,28 @@ json TransformerInstance::extracted_callback(const inja::Arguments &args) const
   return "";
 }
 
+json TransformerInstance::data_source_callback(const inja::Arguments &args) const {
+  const auto& ctx = tls_.getTyped<ThreadLocalTransformerContext>();
+  const std::string &name = args.at(0)->get_ref<const std::string &>();
+  const auto value_it = ctx.data_sources_->find(name);
+  if (value_it != ctx.data_sources_->end()) {
+    return value_it->second->data();
+  }
+  return "";
+}
+
+json TransformerInstance::trim_callback(const inja::Arguments &args) const {
+  std::string s = args.at(0)->get_ref<const std::string &>();
```

Comment:
any reason to not use absl? https://github.com/abseil/abseil-cpp/blob/acf3390ca28edf1438fa896602ffede2a7dff103/absl/strings/ascii.h#L231

Reply:
didn't know absil had this!

Reply:
So it looks like since c++ 17 it is safe to rely on std::isstring but upstream does use the ascii whitspace stripper so it does feel like a nicer strategy imo.
That being said i am fine either way.